app/models/offer.rb

```diff
@@ -259,6 +265,16 @@ def offers_count_per_state(offer)
     end
   end
 
+  def expire_shareable_resource(expires_at=Time.now)
```

Comment:
This is a useful method, but I don't think it should be on the offer. Plus, we'll probably need it for other models in the future.

Can we move it to a static Shareable method ?

e.g

```ruby
Shareable.expire(offer, time)
Shareable.expire(offer.packages, time)
```